### PR TITLE
[nnbd] Move repository to its own Flutter project

### PR DESCRIPTION
**Pull request by CaseyHillers**
With NNBD, Cocoon has reached some issues with being compatible with Flutter. As a temporary workaround, we've been able to move to the beta channel to not cause the breakage.

This PR moves the repository dashboard that is based on a Flutter web fork into its own Flutter project.

## app
- Remove repository logic
- Remove unused json objects (used by old build dashboard)
- Remove json serializable option that was removed
- Remove Flutter for web fork dependencies
- Upgrade pubspec to be nullsafe compatible
- Fixed Angular dart nullability issue

## repo_dashboard
- Copy of the repository files here
- Ran `flutter create`
- Skipped a test that requires some refactoring of the network image code
- Switch use of `dart:html` http calls to `package:http`
- Switch use of `dart:html` localstorage to `package:localstorage`
- Switch use of `dart:html` window.open to `package:url_launcher`
- Switch use of `dart:html` refresh to an HTML attribute
- Switch `print` calls to `debugPrint` to minimize noise in tests
- Fix lints

## app_dart
- To prevent build artifact overlap, repository is moved to `/repository/`, added redirect

## build + test
- `dev/deployment.dart` calls the scripts in `cloudbuild` to reduce build logic duplication
- Deployment: Added repo_dashboard logic to `cloudbuild`
- CI: Add repository to `tests.yaml` based on existing Flutter testing

# Issue

https://github.com/flutter/flutter/issues/70428

# Preview

http://testchillers.flutter-dashboard.appspot.com/repository/index.html

# Future work

I worked on fixing the repo dashboard to work on Flutter, but didn't implement best practices. Additionally, for trivial test failures I disabled them. These skipped tests were because the widget is now off screen due to using a mobile layout instead of the default browser size.

Next steps are to start picking this apart and importing it into the main flutter dashboard app.

### CODE REVIEW

**Comment by CaseyHillers** [comment]
\cc @christopherfujino I have this running and can clean it up today.

My TODO is:
1. Run license adder to repository project
2. Add repository project to Cocoon CI
3. Add repository project to deploy process
4. Fix the roll test as it fails hard with `dart:html` imports

**Review by christopherfujino — COMMENTED** [review]
While we're moving things can we change the repository names? they're very confusing. maybe something like:

app -> benchmarks_dashboard
repository -> repo_dashboard
app_dart -> server
app_flutter -> frontend

**Comment by CaseyHillers** [comment]
> While we're moving things can we change the repository names?

I'm happy to follow up in a separate PR with these renames (outside of repository). Let's keep the scope of this PR to the repository dashboard.

**Comment by christopherfujino** [comment]
> > While we're moving things can we change the repository names?
> 
> I'm happy to follow up in a separate PR with these renames (outside of repository). Let's keep the scope of this PR to the repository dashboard.

makes sense. good chance renaming them will cause an integration test in another repo to fail :P

File: app/pubspec.yaml [hunk 1]
```diff
@@ -1,42 +1,25 @@
 name: cocoon
-version: 0.0.2
 homepage: https://github.com/flutter/cocoon
+publish_to: none
```
Comment (christopherfujino) [inline app/pubspec.yaml:3]: nice lol

**Review by christopherfujino — APPROVED** [review]
LGTM. Thanks for making this change!

**Comment by CaseyHillers** [comment]
@jmagman would you mind reviewing the dashboard changes?

**Comment by CaseyHillers** [comment]
I still need to update the licenses to match the expected format (using `addlicense` wasn't sufficient).

Fix a small issue where going to http://testchillers.flutter-dashboard.appspot.com/repository/#/ 404's (should redirect to the index.html)

**Review by jmagman — COMMENTED** [review]
Let's delete all the `repo_dashboard/*` platform directories we know don't work (desktop, right?)

**Comment by CaseyHillers** [comment]
> Let's delete all the repo_dashboard/* platform directories we know don't work (desktop, right?)

Done.

**Comment by CaseyHillers** [comment]
@keyonghan would you mind reviewing the deploy, cloudbuild, and bin/server.dart changes?

File: app_dart/bin/server.dart [hunk 2]
```diff
@@ -142,13 +142,15 @@ Future<void> main() async {
         final int trimIndex = min(queryIndex, anchorIndex);
         final String filePath = request.uri.path.substring(0, trimIndex);
 
-        const List<String> indexRedirects = <String>['/build.html'];
-        if (indexRedirects.contains(filePath)) {
+        const Map<String, String> redirects = <String, String>{
+          '/build.html': '/#/build',
+          '/repository': '/repository/index.html',
+          '/repository/': '/repository/index.html',
```
Comment (keyonghan) [inline app_dart/bin/server.dart:148]: Do we need these two options? Existing cocoon seems to only reference /repository.html. Or I missed somewhere.

**Review by keyonghan — APPROVED** [review]
LGTM with nit.

File: app_dart/bin/server.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_dart/bin/server.dart:148]: Yes. The Flutter app rewrites `/repository/index.html` to `/repository/#/`. On a reload, Cocoon see this as `/repository` and 404s. Ideally, we'd do something similar to below to just return the `index.html`, but it adds more complexity.

**Comment by CaseyHillers** [comment]
Thanks all for the reviews on such a large PR!

**Pull request merged**